trie/stacktrie.go

```diff
@@ -444,7 +444,7 @@ func (st *StackTrie) hashRec(hasher *hasher, path []byte) {
 	case extNode:
 		st.children[0].hashRec(hasher, append(path, st.key...))
 
-		n := shortNode{Key: hexToCompact(st.key)}
+		n := shortNode{Key: hexToCompactInPlace(st.key)}
```

Comment:
The mutation is on the `st.key` slice directly.

It feels a bit dangerous since https://github.com/ethereum/go-ethereum/blob/master/trie/stacktrie.go#L445 we pass this slice to children for constructing the path. It means, the mutation of parent key will also affect the children.

It can still work because in writerFn, the key and value are deep copied(usually done by database). But I think it's too dangerous just in case we just some other writeFn which doesn't deep-copy key.